Completion tooltip

martindzejky opened the pull request:
I moved the hint under the current line, so it looks like this:

![it1](https://cloud.githubusercontent.com/assets/4700122/13506668/7efa3e3c-e17f-11e5-872f-b71fa52ca3d3.png)

It gets disabled when the code completion appears.

![hint1](https://cloud.githubusercontent.com/assets/4700122/13507330/6ba95a90-e182-11e5-9adc-7a5fe6660c41.png)
![hint2](https://cloud.githubusercontent.com/assets/4700122/13507328/6ba31aa4-e182-11e5-9876-24fa6bd8acfe.png)
![hint3](https://cloud.githubusercontent.com/assets/4700122/13507329/6ba40aae-e182-11e5-904c-e6931b9d7d6b.png)

Closes #1261.
#### UPDATE:

I have added editor settings:

![hint4](https://cloud.githubusercontent.com/assets/4700122/13524137/c4f6b6ac-e1f9-11e5-8769-808ff8026487.png)

And it all works as expected:

![hint5](https://cloud.githubusercontent.com/assets/4700122/13524150/d587056c-e1f9-11e5-9b6d-5e79898311aa.png)
![hint6](https://cloud.githubusercontent.com/assets/4700122/13524151/d587ccd6-e1f9-11e5-8574-0fb03d23dff6.png)

The call hint tooltip gets hidden only when it has to (when to completion is shown on the same side).

martindzejky committed:
Move the call hint under the current line

martindzejky committed:
Hide hint if completion is active

zendorf commented:
Excellent, this has bugged me for a while! I totally agree that the tooltip gets in the way, so this looks like an improvement. Would be great to have a "distance" setting in the editor settings to customise exactly how many pixels above or below the current line it should be.

chuckeles commented:
That's a good point! I'll try to figure out how to add editor settings....

chuckeles commented:
Also see the discussion about how this works in the issue #1261.

martindzejky committed:
Add editor settings for call hint placement

Added settings: text_editor/put_callhint_tooltip_below_current_line
and text_editor/callhint_tooltip_offset

chuckeles commented:
I have add editor settings as you suggested:

![hint4](https://cloud.githubusercontent.com/assets/4700122/13524137/c4f6b6ac-e1f9-11e5-8769-808ff8026487.png)

And it all works as expected:

![hint5](https://cloud.githubusercontent.com/assets/4700122/13524150/d587056c-e1f9-11e5-9b6d-5e79898311aa.png)
![hint6](https://cloud.githubusercontent.com/assets/4700122/13524151/d587ccd6-e1f9-11e5-8574-0fb03d23dff6.png)

The call hint tooltip gets hidden only when it has to (when to completion is shown on the same side).

zendorf commented:
Oh man, that is great work! This should keep everyone happy. Thanks for your efforts :)

akien-mga commented:
That's very nice work!

Could you have a look at your IDE indentation settings? If you check the diffs as shown in GitHub, you'll see that your added code is badly indented. If you could rebase to fix the indentation that would be awesome :)

chuckeles commented:
Hello, my editor is set up to use 2 space for indentation. Looking at the diffs, I guess the rest of the code uses tabs?

akien-mga commented:
> Looking at the diffs, I guess the rest of the code uses tabs?

Indeed.

chuckeles commented:
Now that looks prettier! This is the first time I've done this o.O (rebasing and force pushing)!

chuckeles commented:
Are there any plans on supporting / enforcing / documenting any coding standards?

akien-mga commented:
Currently the coding standards are kind of "write code similar to the existing code". But documenting it all and enforcing it would definitely be a good idea IMO.

akien-mga commented:
Thanks for this nice usability improvement :)
